hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteFunction.java

```diff
@@ -488,11 +505,24 @@ private void flushRemaining(boolean endInput) {
     this.writeStatuses.addAll(writeStatus);
     // blocks flushing until the coordinator starts a new instant
     this.confirming = true;
+
+    writeMetrics.endFlushing();
+    writeMetrics.resetAfterCommit();
+  }
+
+  private void registerMetrics() {
+    MetricGroup metrics = getRuntimeContext().getMetricGroup();
+    writeMetrics = new FlinkStreamWriteMetrics(metrics);
+    writeMetrics.registerMetrics();
   }
 
   protected List<WriteStatus> writeBucket(String instant, DataBucket bucket, List<HoodieRecord> records) {
     bucket.preWrite(records);
-    return writeFunction.apply(records, instant);
+    writeMetrics.startHandleClose();
+    List<WriteStatus> statuses = writeFunction.apply(records, instant);
```

Comment:
What does it mean for handle close?

Reply:
This name is from append write handle close, which is actually file flush. Rename it to `singleFileFlush`,  which make sense in both stream write and append write.

Reply:
Maybe we just rename `checkpointFlush` -> `dataFlush` and `singleFileFlush` -> `fileFlush` ?